src/App/Arguments.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace Friendica\App;
+
+/**
+ * Determine all arguments of the current call, including
+ * - The whole querystring (except the pagename/q parameter)
+ * - The command
+ * - The arguments (C-Style based)
+ * - The count of arguments
+ */
+class Arguments
+{
+	/**
+	 * @var string The complete query string
+	 */
+	private $queryString;
+	/**
+	 * @var string The current Friendica command
+	 */
+	private $command;
+	/**
+	 * @var array The arguments of the current execution
+	 */
+	private $argv;
+	/**
+	 * @var int The count of arguments
+	 */
+	private $argc;
+
+	public function __construct(string $queryString = '', string $command = '', array $argv = [Module::DEFAULT], int $argc = 1)
+	{
+		$this->queryString = $queryString;
+		$this->command     = $command;
+		$this->argv        = $argv;
+		$this->argc        = $argc;
+	}
+
+	/**
+	 * @return string The whole query string of this call
+	 */
+	public function getQueryString()
+	{
+		return $this->queryString;
+	}
+
+	/**
+	 * @return string The whole command of this call
+	 */
+	public function getCommand()
+	{
+		return $this->command;
+	}
+
+	/**
+	 * @return array All arguments of this call
+	 */
+	public function getArgv()
+	{
+		return $this->argv;
+	}
+
+	/**
+	 * @return int The count of arguments of this call
+	 */
+	public function getArgc()
+	{
+		return $this->argc;
+	}
+
+	/**
+	 * Returns the value of a argv key
+	 * @todo there are a lot of $a->argv usages in combination with defaults() which can be replaced with this method
+	 *
+	 * @param int   $position the position of the argument
+	 * @param mixed $default  the default value if not found
+	 *
+	 * @return mixed returns the value of the argument
+	 */
+	public function get(int $position, $default = '')
+	{
+		return $this->has($position) ? $this->argv[$position] : $default;
+	}
+
+	/**
+	 * @param int $position
+	 *
+	 * @return bool if the argument position exists
+	 */
+	public function has(int $position)
+	{
+		return array_key_exists($position, $this->argv);
+	}
+
+	/**
+	 * Determine the arguments of the current call
+	 *
+	 * @param array $server The $_SERVER variable
+	 * @param array $get    The $_GET variable
+	 *
+	 * @return Arguments The determined arguments
+	 */
+	public function determine(array $server, array $get)
```

Comment:
Shouldn't this be in the constructor? It made sense for static classes to have `init` or `determine` methods but do we still need them with class instances?